Realm/RLMSyncPermissionResults.h

```diff
@@ -0,0 +1,109 @@
+////////////////////////////////////////////////////////////////////////////
+//
+// Copyright 2017 Realm Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+// http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+////////////////////////////////////////////////////////////////////////////
+
+#import <Foundation/Foundation.h>
+
+#import "RLMSyncUser.h"
+#import "RLMRealm.h"
+
+NS_ASSUME_NONNULL_BEGIN
+
+/// Properties which a sync permission results collection can be sorted by.
+typedef NS_ENUM(NSUInteger, RLMSyncPermissionResultsSortProperty) {
+    /// Sort by the path of the Realm the permission pertains to.
+    RLMSyncPermissionResultsSortPropertyPath,
+    /// Sort by the user ID of the user the permission applies to.
+    RLMSyncPermissionResultsSortPropertyUserID,
+};
+
+@class RLMSyncPermissionValue;
+
+/**
+ An object representing the results of a permissions query.
+ 
+ This collection will automatically update its contents at the start of each runloop
+ iteration, but the objects it vends are immutable and will themselves not change.
+
+ Permissions results objects are thread-confined, and should not be shared across
+ threads.
+ */
+@interface RLMSyncPermissionResults : NSObject<NSFastEnumeration>
+
+/// The number of results contained within the object.
+@property (nonatomic, readonly) NSInteger count;
+
+/**
+ Return the first permission, or nil if the collection is empty.
+ */
+- (nullable RLMSyncPermissionValue *)firstObject;
```

Comment:
Since this returns a "value type" and not a Realm Object, is `firstObject` the right name for this method? Same goes for `lastObject` and other methods on this type with the word "object" in it.

Also, we translate `firstObject`/`lastObject` to simply `first`/`last` in Swift, and I don't think that's being done here.